### PR TITLE
Mandiant Advantage Integration initial implementation

### PR DESCRIPTION
**Pull request by miguelhdezsob**
## Summary of PR

* Generated the skeleton of the Mandiant Advantage integration package.
* Added data streams.
* Added data collection logic for all the data streams.
* Added the ingest pipeline for all the data streams.
* Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
* Added dashboards and visualizations.
* Added test for pipeline for all the data streams.
* Added system test cases for all the data streams.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Update PR number in changelog once created

## How to test this PR locally

Pull the code locally and run `elastic-package test`. You might also want to build and manually install the integration if needed.

## Related issues

- N/A

## Screenshots

image
image
image

### CODE REVIEW

File: packages/ti_mandiant_advantage/_dev/build/build.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  ecs:
+    reference: git@8.7
```
Comment (kgeller) [inline packages/ti_mandiant_advantage/_dev/build/build.yml:3]: Could we update this to the latest? After today it'll be 8.11

File: packages/ti_mandiant_advantage/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+format_version: 2.5.1
```
Comment (kgeller) [inline packages/ti_mandiant_advantage/manifest.yml:1]: Could we upgrade this to 3.0.0? That is the new standard version for our integrations

File: packages/ti_mandiant_advantage/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        vars:
+          - name: enable_request_tracer
+            type: bool
+            title: Enable request tracing
+            multi: false
+            required: false
+            show_user: true
+            description: The request tracer logs requests and responses to the agent's local file-system for debugging configurations. Enabling this request tracing compromises security and should only be used for debugging. See [documentation](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-httpjson.html#_request_tracer_filename) for details.
+owner:
+  github: elastic/security-external-integrations
```
Comment (kgeller) [inline packages/ti_mandiant_advantage/manifest.yml:50]:
```suggestion
owner:
  github: elastic/security-external-integrations
  type: partner
```

**Comment by kgeller** [comment]
```
Error: there is no owner for "packages/ti_mandiant_advantage" in ".github/CODEOWNERS"
--
  | 🚨 Error: The command exited with status 1
```

Currently the build is failing because there's no entry in [codeowners](https://github.com/elastic/integrations/blob/main/.github/CODEOWNERS). Once that gets added, we should get a 'real' run of CI

**Comment by miguelhdezsob** [comment]
> ```
> Error: there is no owner for "packages/ti_mandiant_advantage" in ".github/CODEOWNERS"
> --
>   | 🚨 Error: The command exited with status 1
> ```
> 
> Currently the build is failing because there's no entry in [codeowners](https://github.com/elastic/integrations/blob/main/.github/CODEOWNERS). Once that gets added, we should get a 'real' run of CI

Thanks for the review! I believe I have addressed your comments and added the CODEOWNERS entry. This should be ready for another review.

File: packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -0,0 +1,399 @@
+---
+description: Pipeline for processing Mandiant IOCs.
+processors:
+- rename:
+      field: message
+      target_field: event.original
+      ignore_missing: true
+- set:
+      field: ecs.version
+      value: 8.7.0
```
Comment (kgeller) [inline packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml:10]:
```suggestion
      value: 8.11.0
```

File: packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -0,0 +1,399 @@
+---
+description: Pipeline for processing Mandiant IOCs.
+processors:
+- rename:
+      field: message
+      target_field: event.original
+      ignore_missing: true
```
Comment (kgeller) [inline packages/ti_mandiant_advantage/data_stream/threat_intelligence/elasticsearch/ingest_pipeline/default.yml:7]:
```suggestion
- rename:
      field: message
      target_field: event.original
      ignore_missing: true
      if: 'ctx.event?.original == null'
```

This if check just helps us prevent conflict in the event the document already has the field set ie when sending from Logstash

**Comment by kgeller** [comment]
@miguelhdezsob CI is looking good. Just a couple final comments and we should be good to go :)

File: packages/ti_mandiant_advantage/data_stream/threat_intelligence/agent/stream/httpjson.yml.hbs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  - delete:
+      target: header.User-Agent
+  - set:
+      # NOTE: The "Build" version must be kept in sync with this package's version.
+      target: header.User-Agent
+      value: '[[userAgent "(Elastic Integration; Mandiant Advantage; Build/0.1.0)"]]'
+  - set:
+      # NOTE: The version must be kept in sync with this package's version.
+      target: header.X-App-Name
+      value: '[["MA-Elastic-Integration-0.1.0)"]]'
```
[the remark this answers is not in the document]
Comment (miguelhdezsob) [reply packages/ti_mandiant_advantage/data_stream/threat_intelligence/agent/stream/httpjson.yml.hbs:21]: It is not a hard requirement, but we do have a best practice of including   the version in this header for any of our integrations. This is used for usage tracking and troubleshooting on our end.

I did leave a comment saying this needs to be manually kept in sync, but could remove the version if you insist.

[the remark this answers is not in the document]
Comment (kgeller) [reply packages/ti_mandiant_advantage/data_stream/threat_intelligence/agent/stream/httpjson.yml.hbs:21]: We can leave it as is. I found out that we do this in a couple of other places as well.

**Review by kgeller — APPROVED** [review]
LGTM

**Pull request merged**